enhancements/microshift/microshift-updateability-ostree.md

```diff
@@ -275,18 +274,17 @@ we will incorporate a mechanism to block specific upgrades by listing version
 numbers _from which_ a new version cannot be upgraded (X.Y+1.Z may include
 X.Y.Z in its "block" list).
 
-If migration is blocked or fails, MicroShift cluster won't start.
-This will render system unhealthy and (if data migration doesn't succeed after
-greenboot's reboot) cause system to roll back.
+If start is blocked because of version skew, MicroShift cluster won't start.
+If migration fails, healthcheck script will detect that and deem system unhealthy.
```

Comment:
Do we want this to still be the case for migrations @dhellmann ?

Reply:
Would we catch errors from the application not being healthy if we didn't check the storage migration explicitly?

Reply:
There shouldn't be an error on the application side unless the previous resource version was gone completely, 3rd party operators  might be the only exception where they don't take care to do proper migration from versions but the users of those operators would need to follow the guidance on upgrading for any 3rd party operator. Furthermore, since we don't allow upgrading MicroShift too far out, that would stop situations where the user hasn't upgraded in a long while, then suddenly jumping to a newer version of MicroShift where an internal resource might not exist anymore.

Reply:
Since this is a "sync doc with what's already implemented" PR and that information wasn't correct I removed it.

I think the only scenario we're vulnerable here is if migration failed, system was healthy, and someone upgraded to a new minor version when version of a Resource was already removed?
Feels very unlikely and more like admin's error (because the platform wasn't put together correctly (missing migration for a CR)), though I don't like we're "healthy" while migration might've failed.

For Resources we ship, I think we should be safe testing upgrades in CI.
For 3rd party apps and CRs, customer should make sure to prepare a robust deployment